cdc/generation.cc

```diff
@@ -59,14 +59,57 @@ static void copy_int_to_bytes(int64_t i, size_t offset, bytes& b) {
     std::copy_n(reinterpret_cast<int8_t*>(&i), sizeof(int64_t), b.begin() + offset);
 }
 
-stream_id::stream_id(int64_t first, int64_t second)
+static constexpr auto stream_id_version_bits = 4;
+static constexpr auto stream_id_random_bits = 32;
+static constexpr auto stream_id_index_bits = sizeof(uint64_t)*8 - stream_id_version_bits - stream_id_random_bits;
+
+static constexpr auto stream_id_random_shift = 0;
+static constexpr auto stream_id_index_shift = stream_id_random_shift + stream_id_random_bits;
+static constexpr auto stream_id_version_shift = stream_id_index_shift + stream_id_index_bits;
+
+/**
+ * Responsibilty for encoding stream_id moved from factory method to 
+ * this constructor, to keep knowledge of composition in a single place. 
+ * Note this is private and friended to topology_description_generator, 
+ * because he is the one who defined the "order" we view vnodes etc. 
+ */ 
+stream_id::stream_id(dht::token token, size_t vnode_index)
     : _value(bytes::initialized_later(), 2 * sizeof(int64_t))
 {
-    copy_int_to_bytes(first, 0, _value);
-    copy_int_to_bytes(second, sizeof(int64_t), _value);
+    static thread_local std::mt19937_64 rand_gen(std::random_device().operator()());
+    static thread_local std::uniform_int_distribution<uint32_t> rand_dist(std::numeric_limits<uint32_t>::max());
```

Comment:
```
rand_dist(std::numeric_limits<uint32_t>::max())
```

That's a singleton distribution.
https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/uniform_int_distribution

Reply:
Hah! What a snafu! Thanks for spotting.

Reply:
I'm gonna trust you to run Scylla a few times and look that at the stream IDs to see that they are generated according to our expectations :)

Reply:
Yup. Manually verified that `rand` is indeed somewhat more distributed.